File: 647-palindromic-substrings/647-palindromic-substrings.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        def countHelper(s, left, right):
            currCount = 0
            while left >= 0 and right < len(s) and s[left] == s[right]:
                left -= 1
                right += 1
                currCount += 1
            return currCount
        count = 0
        for left in range(len(s)):
            count += countHelper(s, left, left)
            count += countHelper(s, left, left + 1)
        return count
        

File: 647-palindromic-substrings/647-palindromic-substrings.java
class Solution {
    private int ans = 0;
    
    public int countSubstrings(String s) {
        if (s == null || s.length() == 0) return 0;
        
        int n = s.length();
        for (int i = 0; i < n; i++) {
            countAux(s, i, i);
            countAux(s, i, i + 1);
        }
        return ans;
    }
    
    public void countAux(String s, int left, int right) {
        while (left >= 0 && right < s.length() && s.charAt(left) == s.charAt(right)) {
            ans++;
            left--;
            right++;
        }
    }
}